zio-http/src/main/scala/zio/http/QueryParams.scala

```diff
@@ -88,25 +89,62 @@ final case class QueryParams(map: Map[String, Chunk[String]]) {
    */
   def getAll(key: String): Option[Chunk[String]] = map.get(key)
 
+  /**
+   * Retrieves all typed query parameter values having the specified name.
+   */
+  def getAllAs[A](key: String)(implicit codec: TextCodec[A]): Either[QueryParamsError, Chunk[A]] =
+    map.get(key) match {
+      case Some(params) =>
+        params
+          .map(param => codec.decode(param).toRight(QueryParamsError.Malformed(key, param, codec)))
+          .partitionMap(identity) match {
+          case (errors, _) if errors.nonEmpty => Left(QueryParamsError.MultiMalformed(errors))
+          case (_, typedParams)               => Right(typedParams)
+        }
+      case None         => Left(QueryParamsError.Missing(key))
+    }
+
   /**
    * Retrieves the first query parameter value having the specified name.
    */
   def get(key: String): Option[String] = getAll(key).flatMap(_.headOption)
 
+  /**
+   * Retrieves the first typed query parameter value having the specified name.
+   */
+  def getAs[A](key: String)(implicit codec: TextCodec[A]): Either[QueryParamsError, A] = for {
```

Comment:
I am wondering, if a `def getAsZIO[A](key: String)(implicit codec: TextCodec[A]): IO[QueryParamsError, A]` makes sense too.

Reply:
Isn't wrapping it in ZIO to heavy machinery for this task? I believe calling it like `ZIO.fromEither(queryParams.getAs ...)` is not much trouble, and current version could be possibly used in some lower level implementations :)

Reply:
I believe there could be a significant gain in readability at the cost of a minor performance overhead. Keeping `getAs` as is and adding `getAsZIO` as a variant seems consistent with existing ZIO functions like `ZIO.serviceWith` and `ZIO.serviceWithZIO`.

Reply:
Thanks for explaining :) You are right, I will add additional functions